src/overlays/actors/ovl_En_Bal/z_en_bal.c

```diff
@@ -64,108 +165,999 @@ static DamageTable D_80A63CE0 = {
+                //  so I am not sure why this is here?
+                this->inflateEarly = true;
+            }
+        } else {
+            this->watchTarget = EN_BAL_WATCHTRG_FAIRY;
+            Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, EN_BAL_ANIM_LAND);
+            Message_StartTextbox(play, 0x1D00, &this->picto.actor);
+            this->lastMessageId = 0x1D00;
+
+            if (this->location == EN_BAL_LOC_CLOCK_TOWN) {
+                this->inflateEarly = true;
+            }
+        }
+        EnBal_SetupTalk(this);
+    } else {
+        if ((this->picto.actor.xzDistToPlayer < 100.0f) && (this->actionFunc != EnBal_InflateBalloon)) {
+            if (this->idleAnimStage != EN_BAL_IDLESTAGE_WAIT) {
+                this->watchTarget = EN_BAL_WATCHTRG_PLAYER;
+            } else {
+                this->watchTarget = EN_BAL_WATCHTRG_NONE;
+            }
+
+            if (!(player->stateFlags1 & PLAYER_STATE1_800000) && !(player->actor.bgCheckFlags & BGCHECKFLAG_WATER) &&
+                ((this->timer < 300) || (this->timer == 301))) {
+                func_800B8614(&this->picto.actor, play, 100.0f);
+            }
+        } else {
+            this->watchTarget = EN_BAL_WATCHTRG_NONE;
+        }
+
+        if (Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame)) {
+            if (this->idleAnimStage == EN_BAL_IDLESTAGE_PREP_WAIT) {
+                this->forceEyesShut = false;
+                this->eyeTexIndex = EN_BAL_EYETEX_OPEN;
+                Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, EN_BAL_ANIM_WAIT);
+                this->idleAnimStage++;
+            } else if (this->idleAnimStage == EN_BAL_IDLESTAGE_WAIT) {
+                if (Rand_Next() & 1) {
+                    this->forceEyesShut = false;
+                    this->eyeTexIndex = EN_BAL_EYETEX_OPEN;
+                    Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, EN_BAL_ANIM_TALK_IDLE);
+                } else {
+                    this->forceEyesShut = true;
+                    Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, EN_BAL_ANIM_SPIN);
+                }
+                this->idleAnimStage = EN_BAL_IDLESTAGE_ACTIVITY;
+            } else {
+                this->idleAnimStage++;
+            }
+        } else if ((this->idleAnimStage == EN_BAL_IDLESTAGE_WAIT) && (Animation_OnFrame(&this->skelAnime, 20.0f))) {
+            this->forceEyesShut = true;
+        }
+    }
+}
+
+void EnBal_SetupTalk(EnBal* this) {
+    this->timer = 0;
+    this->actionFunc = EnBal_Talk;
+}
+
+void EnBal_Talk(EnBal* this, PlayState* play) {
+    Player* player;
+
+    player = GET_PLAYER(play);
+    player->stateFlags2 |= PLAYER_STATE2_100000;
+    this->isTalking = false;
+
+    switch (Message_GetState(&play->msgCtx)) {
+        case TEXT_STATE_NONE:
+        case TEXT_STATE_1:
+            break;
+        case TEXT_STATE_CLOSING:
```

Comment:
```suggestion
            break;

        case TEXT_STATE_CLOSING:
```
Can you add new lines between breaks and new cases? It helps separate the cases nicely. Same for other switches.